dateparser/date.py

```diff
@@ -214,19 +217,21 @@ def _is_valid_date_obj(self, date_obj):
 
 class DateDataParser(object):
 
-    def __init__(self, language=None, allow_redetect_language=False):
-        if isinstance(language, basestring):
+    def __init__(self, languages=None, allow_redetect_language=False):
+        if isinstance(languages, collections.Iterable):
```

Comment:
It should check for `list` instance instead. I missed the fact that even strings are `Iterable`. Fixed it and also added else clause to raise TypeError in case invalid type is supplied.